build/linux/ncpa.spec

```diff
@@ -52,23 +52,23 @@ rm -rf %{buildroot}
 %pre
 if which chkconfig &> /dev/null; then
     echo "Try to stop services with chkconfig"
-    /usr/local/ncpa/ncpa_listener --stop &> /dev/null
-    /usr/local/ncpa/ncpa_passive --stop &> /dev/null
+    [ -f /usr/local/ncpa/ncpa_listener ] && /usr/local/ncpa/ncpa_listener --stop &> /dev/null
+    [ -f /usr/local/ncpa/ncpa_passive ] && /usr/local/ncpa/ncpa_passive --stop &> /dev/null
     chkconfig --del ncpa_listener
     chkconfig --del ncpa_passive
 fi
 if command -v systemctl &> /dev/null
 then
     echo "Try to stop services with systemctl"
-    systemctl stop ncpa_listener &> /dev/null || true
-    systemctl stop ncpa_passive &> /dev/null || true
+    systemctl list-units --full -all | grep -Fq 'ncpa_listener.service' && systemctl stop ncpa_listener &> /dev/null || true
```

Comment:
Add corresponding conditional deletes of services for the systemctl and service cases